src/v/cloud_storage/materialized_resources.cc

```diff
@@ -75,6 +202,68 @@ materialized_resources::materialized_resources()
             return _manifest_cache->set_capacity(_manifest_meta_size());
         });
     });
+
+    auto reset_tp = [this] {
+        ssx::spawn_with_gate(_gate, [this] { return update_throughput(); });
+    };
+
+    _throughput_shard_limit_config.watch(reset_tp);
+    _relative_throughput.watch(reset_tp);
+
+    reset_tp();
+}
+
+ss::future<> materialized_resources::update_throughput() {
+    auto tp = get_throughput_limit(_device_throughput);
+    if (ss::this_shard_id() == 0) {
+        co_await set_disk_max_bandwidth(tp.disk_node_throughput_limit);
+    }
+    set_net_max_bandwidth(tp.download_shard_throughput_limit);
+}
+
+void materialized_resources::set_net_max_bandwidth(size_t tput) {
+    if (tput != 0 && tput != std::numeric_limits<size_t>::max()) {
+        vlog(
+          cst_log.info,
+          "Setting cloud storage download bandwidth to {} on this shard",
+          tput);
+        _throughput_limit.update_capacity(tput);
+        _throughput_limit.update_rate(tput);
+        _throttling_disabled = false;
+    } else {
+        vlog(
+          cst_log.info,
+          "Disabling cloud storage download throttling on this shard");
+        _throttling_disabled = true;
+    }
+}
+
+ss::future<> materialized_resources::set_disk_max_bandwidth(size_t tput) {
+    try {
+        if (tput == 0 || tput == std::numeric_limits<size_t>::max()) {
+            _throttling_disabled = true;
```

Comment:
`_throttling_disabled` only tracks whether network based throttling should be done. At least that's how it's used. Doesn't make much sense to update it here.

Reply:
will update in a followup